### PR TITLE
Don't remove `node_modules` in `@redwoodjs` packages on `yarn rwfw project:copy`

### PR DESCRIPTION
**Pull request by jtoar**
Closes https://github.com/redwoodjs/redwood/issues/4937.

TL;DR yarn does a lot of work to make `node_modules` exactly the way it should be, so we shouldn't change it. 

When we run `yarn rwfw project:copy`, we `rimraf` all the `@redwoodjs` packages in a Redwood project's `node_modules`: (line 109) 

https://github.com/redwoodjs/redwood/blob/3dc54a6dc113b8b38c619366d584e98174d9c420/tasks/framework-tools/lib/project.mjs#L103-L109

To make this more concrete, for `@redwoodjs/testing`, `packageDistPath` ends up being `"/Users/dom/prjcts/rwfw/rwfw/node_modules/@redwoodjs/testing"` (I named my Redwood project "rwfw"). Here's what the directory structure of that filepath looks like:

```
/Users/dom/prjcts/rwfw/rwfw/node_modules/@redwoodjs/testing/
├── node_modules/
│  ├── mime/
│  ├── // so, many, more...
│  └── acorn-jsx/
├── web/
│  ├── package.json
│  └── index.js
├── api/
│  ├── package.json
│  └── index.js
├── dist/
│  ├── web/
│  ├── api/
│  └── tsconfig.tsbuildinfo
├── config/
│  ├── storybook/
│  └── jest/
├── README.md
├── package.json
└── LICENSE
```

`yarn rwfw project:copy` wants to replace everything in this directory with files built from your local copy of the framework. This is what it ends up looking like after it does so:

```
/Users/dom/prjcts/rwfw/rwfw/node_modules/@redwoodjs/testing/
├── web/
│  ├── package.json
│  └── index.js
├── api/
│  ├── package.json
│  └── index.js
├── dist/
│  ├── web/
│  ├── api/
│  └── tsconfig.tsbuildinfo
├── config/
│  ├── storybook/
│  └── jest/
├── README.md
└── package.json
```

Notice that `node_modules` is gone (and LICENSE too, but that doesn't matter). This may seem innocuous, but it causes problems if you run `yarn install` again because yarn doesn't expect the contents of `node_modules` to have changed from under it. Especially the contents of a package's `node_modules` in `node_modules`.

Instead, when running `yarn rwfw project:copy`, let's just leave `node_modules` in `@redwoodjs` packages the way they are.

### CODE REVIEW

**Comment by Tobbe** [comment]
In your example you have `acorn-jsx` inside `node_modules`. What happens if one of the changes I'm trying to test depends on a new version of `acorn-jsx`? If we just leave that one in there, won't my change still be using the old version after I've run project:copy to get it over to my test project?

**Comment by jtoar** [comment]
As long as I'm understanding your question correctly I think that case is handled. If you're testing a new version of a dependency, say `acorn-jsx`, and you add that dependency to the framework (say that it's in `@redwoodojs/web`'s `package.json` now), it should get accounted for in the `yarn rwfw project:deps` step:

```bash
yarn rwfw project:deps
yarn install
# Ok, dependencies / node_modules structure are all set.
# Now we _just_ need the files.
yarn rwfw project:copy
```

But the smoke test is failing so maybe by theory is all wrong. 😅 Lets see what's up with that.

**Comment by Tobbe** [comment]
You got it :)  I just didn't know about `project:deps`. Should we add that to our contributing docs? Unless I'm missing something when reading the code it looks like right now it's only used by the test-project and our e2e tests

**Comment by jtoar** [comment]
Linking this issue I opened up in the yarn repo for housekeeping purpose: https://github.com/yarnpkg/berry/issues/4429.

**Comment by jtoar** [comment]
The strategy in this PR doesn't seem viable. Moreover, Yarn v3.2.1 seems to fix this!
